### DISCUSSION TITLE
X1-Boost, X1-Mini, X1-Air modbus documentation

### DISCUSSION THREAD

**Discussion by Goblaski** [wills106/homeassistant-solax-modbus#327] [General]
I've gotten the public Modbus protocol info for the X1-Boost, Mini and air. Any use for it? If so I can share it. 

I've so far have not been able to get the X1-Boost I have at home working, but I will try with a shorter cable first. If it works I will let you know.

**Comment by mapotn** [comment]
hey, do you have working integration with solax x1 boost? I have x1 boost gen3 + wifi dongle v3 but I am not able to use this integration yet. thanks
![image](https://user-images.githubusercontent.com/50015021/218341223-00c5ad49-3bf6-40eb-a9a5-534b301d6cf6.png)

**Reply by wills106** [reply]
You are best contacting SolaX to ask if the X1 Boost range with Modbus support are physically a different product, or if non-Modbus models can work with a firmware change.

**Reply by mapotn** [reply]
Well my inverter is already comunicating via RS485 modbus with Eastron smartmeter

**Reply by mapotn** [reply]
Answer from solax support: The firmware of communication module on your inverter is already the latest one and it is capable for modbus communication.

**Reply by wills106** [reply]
Can you get access to one of those cheep RS485 - USB adaptors to rule out the Pocket WiFi?

**Reply by mapotn** [reply]
Which variation? FT232RL 
CH340G 
PL2303

**Reply by wills106** [reply]
It's been used on the really Cheap CH340 you can get for a couple of £ / € or the dearer DSD Tech devices based on CP2102
There are a couple of links in the Wiki for the cheaper option, so you know what to look for.

**Reply by Goblaski** [reply]
> Answer from solax support: The firmware of communication module on your inverter is already the latest one and it is capable for modbus communication.

Which version do you have. I have Master 1.08 and Manager 1.07.

**Reply by wills106** [reply]
Looks like Arm 1.41 and D2.22 https://github.com/wills106/homeassistant-solax-modbus/discussions/26#discussioncomment-4951225

According that that doc you need Arm 1.37

**Reply by mapotn** [reply]
should I downgrade to 1.37?

**Reply by Goblaski** [reply]
I don't think so. I am currently on ARM 1.46 now and it is working.

**Reply by wills106** [reply]
Are you still trying to connect with a Pocket WiFi or an RS485 adaptor?

**Reply by mapotn** [reply]
@Goblaski so finally after upgrade to 1.46 integration works? if yes i will ask support to upgrade my invertor's fw to latest one - 1.46

**Reply by Goblaski** [reply]
Yes but do note that I was on 1.36 before (and at least 1.37 was supported). As Wills mentioned it is unsure if Modbus of WiFi is possible.

**Reply by wills106** [reply]
I gave a better answer regarding Pocket WiFi here https://community.home-assistant.io/t/universal-solar-inverter-over-modbus-rs485-tcp-custom-component-formally-solax-no-pocket-wifi/140143/1014?u=wills106

Also to add to that comment. While each Gen of the Hybrid Inverters share the majority of the same registers, there are differences. That could be why they only mention the Gen4 Hybrids as working. It might be how they map the Modbus to the WiFi Stick.

**Reply by mapotn** [reply]
thanks for checking it in another thread, i will try upgrade my fw to latest or ill buy RS485 adaptor soon. One question, if I will replace wifi dongle with RS485 adaptor will be still working Solax cloud app ?

**Reply by wills106** [reply]
You can use both at the same time as they connect to a different port.

**Reply by mapotn** [reply]
Wanted to try RS485-usb adapter and my inverter has only one usb which is used by wifi dongle currently and second is for upgrade only. (looks like I dont have LAN connector)

**Reply by wills106** [reply]
With a USB-RS485 device, the USB end goes in your computer / whatever is running Home Assistant.
The screw terminals are the RS485 end of the adaptor. You will need to connect that to the RS485 connection on the Inverter, which is normally PIN 4 & 5 on one of the RJ45 sockets. But you need to refer to your User Manual for you Inverter to ensure you are connected correctly!

Edit: Have a loot at "Installation Notes" within the Wiki

**Reply by mapotn** [reply]
actually I have smart meter connected in this RJ45, I would lost home/grid consumption info without it ;(

**Reply by wills106** [reply]
You can daisy chain with Modbus.

**Reply by mapotn** [reply]
Do you think I can connect it like this? thanks
![image](https://user-images.githubusercontent.com/50015021/219940376-4d4f52d8-c4fd-4c48-afc0-fee33a0faee7.png)

**Reply by wills106** [reply]
Should be fine.

**Reply by mapotn** [reply]
thanks, i will try it as I have new fw now and still same issue

**Reply by mapotn** [reply]
Hi @wills106 any idea how to set this? Thanks 
![Screenshot_20230219-220855_Home Assistant](https://user-images.githubusercontent.com/50015021/219975393-ee3e0335-93d4-444a-a052-30baf92709e8.png)

**Reply by wills106** [reply]
Baudrate is 9600

**Reply by mapotn** [reply]
unfortunately I have no success even with rs485 USB adaptor, all still looks same in logs etc. 
should I use same pins 4,5 in RJ45 connector? 
![image](https://user-images.githubusercontent.com/50015021/224509703-9d3e858e-890e-4b11-8120-50f68e8f28ce.png)
trying to use /dev/ttyUSB0 path as it looks correct in hw details in HA:

ttyUSB0
/dev/serial/by-id/usb-1a86_USB2.0-Ser_-if00-port0
Subsystém:
tty
Cesta k zariadeniu:
/dev/ttyUSB0
ID:
/dev/serial/by-id/usb-1a86_USB2.0-Ser_-if00-port0
Atribúty:
DEVLINKS: >-
  /dev/serial/by-id/usb-1a86_USB2.0-Ser_-if00-port0
  /dev/serial/by-path/platform-xhci-hcd.0.auto-usb-0:1.1:1.0-port0
DEVNAME: /dev/ttyUSB0
DEVPATH: >-
  /devices/platform/soc/ffe09000.usb/ff500000.usb/xhci-hcd.0.auto/usb1/1-1/1-1.1/1-1.1:1.0/ttyUSB0/tty/ttyUSB0
ID_BUS: usb
ID_MODEL: USB2.0-Ser_
ID_MODEL_ENC: USB2.0-Ser\x21
ID_MODEL_ID: '7523'
ID_PATH: platform-xhci-hcd.0.auto-usb-0:1.1:1.0
ID_PATH_TAG: platform-xhci-hcd_0_auto-usb-0_1_1_1_0
ID_REVISION: '0254'
ID_SERIAL: 1a86_USB2.0-Ser_
ID_TYPE: generic
ID_USB_DRIVER: ch341
ID_USB_INTERFACES: ':ff0102:'
ID_USB_INTERFACE_NUM: '00'
ID_VENDOR: 1a86
ID_VENDOR_ENC: 1a86
ID_VENDOR_ID: 1a86
MAJOR: '188'
MINOR: '0'
SUBSYSTEM: tty
TAGS: ':systemd:'
USEC_INITIALIZED: '4613898'

**Comment by Goblaski** [comment]
Official documentation: [Solax Power X1-BOOST Single phase inverter Modbus-RTU comms protocolV1.0 - Public version.pdf](https://github.com/wills106/homeassistant-solax-modbus/files/10720051/Solax.Power.X1-BOOST.Single.phase.inverter.Modbus-RTU.comms.protocolV1.0.-.Public.version.pdf)

This seems to be Modbus @wills106. For my case I expect a dodgy cable at the moment, but I will try a with short cable and a laptop next to the inverter some time next two weeks to see if it works. My current UTP cable is about 20-30 meters long and running through the house. Might have damaged it during install so I will try a short one first.

**Reply by wills106** [reply]
That is the Modbus document. But there is also a non Modbus document available.
I don't know if all of these Inverters come with non Standard protocol over RS485 and you can request the Modbus Firmware from SolaX. Or is there is a separate Hardware version supporting Modbus?

**Reply by michaeljaddy** [reply]
@Goblaski Did you manage to get this working on the X1-Boost?

**Reply by Goblaski** [reply]
Yes, worked directly after a firmware update.

**Reply by michaeljaddy** [reply]
Great thanks! Which modbus address and baud rate did you use?

**Reply by Goblaski** [reply]
Well after the update you will need to set the modbus to enabled in your Admin Menu on the Inverter. You can set the address over there and the baud I think was 9600. I used the default address at the time (1).

**Reply by michaeljaddy** [reply]
Thank you. Was the rs485 enabled in the "DRM function" menu?

**Reply by michaeljaddy** [reply]
These look like the menu options I have available to me:

![image](https://github.com/wills106/homeassistant-solax-modbus/assets/113230724/bacd02c9-f259-447c-8080-21b30d13c04e)

**Reply by Goblaski** [reply]
After the update with the right firmware there should be an extra option for Modbus. I can’t recall if it was in the DRM menu or next to it. Just make sure you update first in order to have the submenu appear.

**Reply by michaeljaddy** [reply]
Ahh okay apologies I didn't realise the setting appeared as a direct result of the firmware upgrade. May I ask what firmware you are running now?

My inverter shows:
Master: 1.08
Manager: 1.07

If this means anything to you?

**Reply by michaeljaddy** [reply]
Oh also:
Internal code:
A: 1.38 D: 2.22

**Reply by Goblaski** [reply]
Firmware Arm 1.37 and above is supported. So technically it should support ModBus. I think I got firmware ARM 1.41 (recieved the firmware from Solax). So if it doesn’t show up in the menu you might want to get the the latest firmware. I am unfortunately unable to access the inverter at the moment so I can’t check the setting.

**Reply by michaeljaddy** [reply]
Perfect thank you! Just waiting for the installer to send over the PIN number to access the settings and will give it a shot.

**Reply by wills106** [reply]
The default is 2014 or 6868 unless they have changed it:

https://github.com/wills106/homeassistant-solax-modbus/blob/a2f970ae4a974aa49e9c88f5d067a6a9be2835db/custom_components/solax_modbus/plugin_solax.py#L919-L930

Edit: Assuming the Air / Boost / Mini is the same pin as the Hybrids.

**Reply by michaeljaddy** [reply]
Perfect thanks for your input folks! Up and running:

![image](https://github.com/wills106/homeassistant-solax-modbus/assets/113230724/5dc9d1d0-faea-4c35-9792-f454a12eb6b1)
